internal/lefthook/run.go

```diff
@@ -124,6 +125,7 @@ Run 'lefthook install' manually.`,
 			SkipSettings:    logSettings,
 			DisableTTY:      cfg.NoTTY || args.NoTTY,
 			AllFiles:        args.AllFiles,
+			FilesFromStdin:  args.FilesFromStdin,
```

Comment:
I think there's no need to pass this argument. You can parse the files here and pass them via `Files` option.

Reply:
If the functionality isn't behind a CLI option, it will break cases where users supply other information to standard input (e.g., using `yes`).

Reply:
Sorry, I don't get it. If the functionality isn't behind a CLI option, then lefthook is called from within a git hook. In this case files from STDIN don't make sense, since git doesn't pass files via STDIN.

And if lefthook is called explicitly, I'd like to make sure that it is called correctly. If the override is required – let's rely on the CLI options.

I want to distinguish the implicit calls that must make sense and explicit calls that tweak lefthook behavior, for instance change the files passed to the commands. That's my vision and I believe it aligns with KISS principle.

Reply:
> Sorry, I don't get it. If the functionality isn't behind a CLI option, then lefthook is called from within a git hook. In this case files from STDIN don't make sense, since git doesn't pass files via STDIN.

First, let's discuss Git hook behavior. According to its docs, Git hook do get information through standard input, see e.g.: https://git-scm.com/docs/githooks#_pre_push

> And if lefthook is called explicitly, I'd like to make sure that it is called correctly. If the override is required – let's rely on the CLI options.

That was exactly my intention, but previously you wrote:
>  I think there's no need to pass this argument.

Reply:
> I think there's no need to pass this argument.

It was related to Go code. I meant it's better to read from STDIN in `internal/lefthook/run.go` and pass the results as `Files` option. It's Ok to have `--files-from-stdin` CLI option that replaces files in lefthook commands.

> Git hooks do get information through standard input

Yes, but it is related to Git while getting files from STDIN is not about git.

Reply:
> Git passes paths via stdin to a hook script, such as Lefthook, in some cases.

Git passes data with a special format: https://git-scm.com/docs/githooks#_pre_push. This is not files list separated with newlines. So, an option `files_from_stdin` can't be safely used for a git hook. In this case this option is only valid for the explicit call of lefthook. This means we can keep just a CLI argument `--files-from-stdin`

Reply:
Let me show how I see the configuration:

```yml
# lefthook.yml

custom-hook:
  commands:
    lint:
      files: git diff --name-only main # custom files
      run: python scripts/linter.py {files}
```

When called like `lefthook run custom-hook`, `{files}` will be replaced with the results of `git diff --name-only main`. When called like `find . -name '*.py' -print0 | lefthook run custom-hook --files-from-stdin` `{files}` will be replaced from the \0-separated list from STDIN.

Does it sound good?

Reply:
@mrexox Only for pre-push, pre-receive and reference-transaction stdin gets used. I was intending for the configuration option to apply even when Lefthook is invoked from the CLI. I prefer check for those hooks that use stdin in the source code rather than not having the config key.